nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestTailFile.java

```diff
@@ -620,6 +620,27 @@ public void testMultipleFiles() throws IOException, InterruptedException {
         runner.getFlowFilesForRelationship(TailFile.REL_SUCCESS).get(4).assertContentEquals("1\n");
     }
 
+    @Test
+    public void testMultipleFilesWithBasedirAndFilenameEL() throws IOException, InterruptedException {
+        runner.setVariable("vrBaseDirectory", "target");
+        runner.setProperty(TailFile.BASE_DIRECTORY, "${vrBaseDirectory}");
+        runner.setProperty(TailFile.MODE, TailFile.MODE_MULTIFILE);
+        runner.setVariable("vrFilename", "(testDir/)?log(ging)?.txt");
+        runner.setProperty(TailFile.FILENAME, "${vrFilename}");
+        runner.setProperty(TailFile.ROLLING_FILENAME_PATTERN, "${filename}.?");
+        runner.setProperty(TailFile.START_POSITION, TailFile.START_CURRENT_FILE);
+        runner.setProperty(TailFile.RECURSIVE, "true");
+
+        runner.run(1);
+        runner.assertAllFlowFilesTransferred(TailFile.REL_SUCCESS, 0);
+
+        otherRaf.write("hi\n".getBytes());
```

Comment:
Just to ensure we don't have weird errors on Windows-based environments consider specifying:
```
"hi\n".getBytes(StandardCharsets.UTF_8);
```